src/Workspaces/Core/Portable/Utilities/ImmutableArrayExtensions.cs

```diff
@@ -39,11 +39,11 @@ internal static ImmutableArray<T> ToImmutableArrayOrEmpty<T>(this IEnumerable<T>
         internal static ImmutableArray<T> ToImmutableArrayOrEmpty<T>(this ImmutableArray<T> items)
             => items.IsDefault ? ImmutableArray<T>.Empty : items;
 
-        internal static IReadOnlyList<T> ToImmutableReadOnlyListOrEmpty<T>(this IEnumerable<T> items)
+        internal static ImmutableArray<T> ToImmutableReadOnlyListOrEmpty<T>(this IEnumerable<T> items)
```

Comment:
:bulb: This method should really be renamed if the fan-out isn't too horrible. I already found it confusing earlier in this review.